server/models/Transaction.js

```diff
@@ -506,5 +510,94 @@ export default (Sequelize, DataTypes) => {
+        transaction.paymentMethodCollectiveId = paymentMethod.CollectiveId;
+        const pmCollective = await models.Collective.findById(paymentMethod.CollectiveId);
+        transaction.paymentMethodCollectiveSlug = pmCollective.slug;
+      }
+    }
+    if (transaction.OrderId) {
+      const order = await models.Order.findById(transaction.OrderId);
+      if (order.FromCollectiveId) {
+        transaction.orderFromCollectiveId = order.FromCollectiveId;
+        const orderFromCollective = await models.Collective.findById(order.FromCollectiveId);
+        transaction.orderFromCollectiveSlug = orderFromCollective.slug;
+      }
+      if (order.PaymentMethodId) {
+        const orderPaymentMethod = await models.PaymentMethod.findById(order.PaymentMethodId);
+        transaction.orderPaymentMethodService = orderPaymentMethod.service;
+        transaction.orderPaymentMethodType = orderPaymentMethod.type;
+        if (orderPaymentMethod.CollectiveId) {
+          transaction.orderPaymentMethodCollectiveId = orderPaymentMethod.CollectiveId;
+          const orderPaymentMethodCollective = await models.Collective.findById(orderPaymentMethod.CollectiveId);
+          transaction.orderPaymentMethodCollectiveSlug = orderPaymentMethodCollective.slug;
+        }
+      }
+    }
+    if (transaction.ExpenseId) {
+      const expense = await models.Expense.findById(transaction.ExpenseId);
+      transaction.expensePayoutMethod = expense.payoutMethod;
+      if (expense.UserId) {
+        transaction.expenseUserId = expense.UserId;
+        const expenseUser = await models.User.findById(expense.UserId);
+        transaction.expenseUserPaypalEmail = expenseUser.paypalEmail;
+        if (expenseUser.CollectiveId) {
+          const expenseUserCollective = await models.Collective.findById(expenseUser.CollectiveId);
+          transaction.expenseUserCollectiveSlug = expenseUserCollective.slug;
+        }
+      }
+      if (expense.CollectiveId) {
+        const expenseCollective = await models.Collective.findById(expense.CollectiveId);
+        transaction.expenseCollectiveId = expense.CollectiveId;
+        transaction.expenseCollectiveSlug = expenseCollective.slug;
+      }
+    }
+    return transaction;
+  };
+
+  Transaction.postTransactionToLedger = async transaction => {
+    if (transaction.deletedAt || transaction.type !== 'CREDIT') {
+      return Promise.resolve();
+    }
+    try {
+      const ledgerTransaction = await getTransactionWithNestedProperties(transaction);
+      if (!process.env.QUEUE_ENABLED) {
+        return axios.post(ledger.transactionUrl, ledgerTransaction);
+      }
+      const conn = await amqp.connect(ledgerQueue.url);
+      const channel = await conn.createChannel();
+      await channel.assertQueue(ledgerQueue.transactionQueue, { exclusive: false });
+      channel.sendToQueue(ledgerQueue.transactionQueue, Buffer.from(JSON.stringify([ledgerTransaction]), 'utf8'));
+    } catch (error) {
```

Comment:
What's the point of catching the error to just throw it?

Reply:
removing it, was logging before but I don't think it's necessary anymore, at least here.

Reply:
@marcogbarcellos I think it's best to keep the `try/catch`, we don't want to throw errors if we have a problem there.

I'm not sure what's the behavior with Sequelize hooks, what happen if they throw an error?

Reply:
Okay, so I'll just log the error. As it's an async method All it did locally was log the error as there was no other method waiting on any answer. But I could instead of throw just `console.error` or `debug` the error if so.